data/ai/lua/ca_recruit_rushers.lua

```diff
@@ -1,50 +1,1091 @@
--- Make the generic_recruit_engine functions work as external CAs
+-- cfg: parameters for configuring recruitment, read from the candidate action's [args] tag
+--    ca_score: score of the candidate action
+--      (default: 196000 as defined in AI_CA_RECRUIT_RUSHERS_SCORE)
+--    randomness: a measure of randomness in recruitment
+--      higher absolute values increase randomness, with values above about 3 being close to completely random
+--      (default = 0.1)
+--    high_level_fraction: the approximate fraction of units of level 2 or higher to be recruited
+--      (default = 0)
+--     enemy_types: array of default enemy unit types to consider if there are no enemies on the map
+--      and no enemy sides exist or have recruit lists
```

Comment:
I just noticed that specifying `enemy_types` as a Lua array does not work (and has never worked) if this parameter is to be supplied by an `[args]` tag. We could convert it to a comma-separated list instead, but I suggest to simply drop this parameter altogether. It is an undocumented parameter and, as is obvious from the comment above, it is essentially never needed, and even in the rare scenario when that might happen, there is a default list of types provided that will be used instead. This is such a rare edge case that it does not seem worth it to me to keep this parameter.

Reply:
As I recall, I added it with maps like Dark Forecast in mind, where there are lots of enemies of known types, but they're all placed via scripting. Humans can play it fine after the first few games, but the AI has no way of knowing what's coming and this would help with the first round of enemies.

Of course, I never actually tried making an AI for Dark Forecast, so I never discovered the issue. I suspect I would have ended up either providing the parameter through Lua and not noticed anyway or modifiing it further to support changing spawn lists.

At any rate, unless you plan on writing a Dark Forecast AI, it seems completely reasonable to drop it.

Reply:
Thanks for the comments.  I am not planning to write an AI specifically for Dark Forecast either, so I am going to stick with dropping this for now.  We can always add something (which would likely end up being more complex than this) later, either in this CA or a custom setup, if the need arises.